### PR TITLE
[DataGridPro] Expose the field of the tree data grouping column as a constant

### LINKED ISSUE #3434: [DataGridPro] Export the tree data grouping column field constant
### Order ID 💳

#26327

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

I found in the documentation how to pin the checkbox to the right using `GRID_CHECKBOX_SELECTION_COL_DEF.field`

```
<DataGridPro
        rows={rows}
        columns={columns}
        checkboxSelection
        initialState={{
          pinnedColumns: {
            left: [GRID_CHECKBOX_SELECTION_COL_DEF.field],
            right: ['actions'],
          },
        }}
      />
```
But I am unable to find COL_DEF for "Group" column using TreeData.

I would like to pin "Job Title" and "Group" column for example.

Bildschirmfoto 2021-12-14 um 16 45 26
https://codesandbox.io/s/wgyrh?file=/demo.tsx

### The problem in depth 🔍

_No response_

### Context 🔦

_No response_

### Your environment 🌎

<details>
  <summary>`npx @material-ui/envinfo`</summary>

```
 System:
    OS: macOS 12.0.1
  Binaries:
    Node: 14.17.2 - /usr/local/bin/node
    Yarn: Not Found
    npm: 7.20.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 96.0.4664.93
    Edge: 96.0.1054.53
    Firefox: 92.0
    Safari: 15.1
  npmPackages:
    @emotion/react: ^11.7.1 => 11.7.1 
    @emotion/styled: ^11.6.0 => 11.6.0 
    @material-ui/styles: ^4.11.1 => 4.11.4 
    @material-ui/types:  5.1.0 
    @material-ui/utils:  4.11.2 
    @types/react: ^17.0.37 => 17.0.37 
    react: ^17.0.2 => 17.0.2 
    react-dom: ^17.0.2 => 17.0.2 
    typescript: ^3.7.5 => 3.9.10 
```
</details>

### PR DESCRIPTION
**Pull request by flaviendelangle**
Fixes #3434

I renamed the column internal variables with `GROUPING` instead of `GROUP` to follow the convention of #212.

### CODE REVIEW

**Review by m4theushw — COMMENTED** [review]
We only miss docs so users can find this option. I was thinking about a new section in https://mui.com/components/data-grid/group-pivot/#tree-data. I don't think a demo is needed, only the code example.

**Comment by flaviendelangle** [comment]
@m4theushw Done
I added it to the `Custom grouping column`

**Review by alexfauquette — APPROVED** [review]
That sounds much better than my idea of exporting `GRID_TREE_DATA_GROUP_COL_DEF_FORCED_PROPERTIES`

File: docs/src/pages/components/data-grid/group-pivot/group-pivot.md [hunk 1]
```diff
@@ -61,6 +61,22 @@ Use the `groupingColDef` prop to customize the rendering of the grouping column.
 
 {{"demo": "pages/components/data-grid/group-pivot/CustomGroupingColumnTreeData.js", "bg": "inline", "defaultCodeOpen": false}}
 
+#### Accessing the grouping column field
+
+If you want to access the grouping column field, for instance to pin it, you can use the `GRID_TREE_DATA_GROUPING_FIELD` constant.
+
+```tsx
+<DataGrid
```
Comment (m4theushw) [inline docs/src/pages/components/data-grid/group-pivot/group-pivot.md:69]:
```suggestion
<DataGridPro
```

File: docs/src/pages/components/data-grid/group-pivot/group-pivot.md [hunk 2]
```diff
@@ -61,6 +61,22 @@ Use the `groupingColDef` prop to customize the rendering of the grouping column.
 
 {{"demo": "pages/components/data-grid/group-pivot/CustomGroupingColumnTreeData.js", "bg": "inline", "defaultCodeOpen": false}}
 
+#### Accessing the grouping column field
+
+If you want to access the grouping column field, for instance to pin it, you can use the `GRID_TREE_DATA_GROUPING_FIELD` constant.
```
Comment (m4theushw) [inline docs/src/pages/components/data-grid/group-pivot/group-pivot.md:66]: Explicitly writing "column pinning" makes easier to find with <kbd>CTRL</kbd> + <kbd>f</kbd>. 

```suggestion
If you want to access the grouping column field, for instance, to use it with column pinning, the `GRID_TREE_DATA_GROUPING_FIELD` constant is available.
```

**Review by m4theushw — APPROVED** [review]

**Pull request merged**